diagnostics-app/src/app/core/services/theme.service.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Global service for managing theme related operations for the app.
+ */
+
+import { EventEmitter, Injectable } from '@angular/core';
+import { Theme } from '../enums/theme.enum';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ThemeService {
+  private _theme: Theme = 'dark';
+
+  // subscribed by app and header
+  // to listen for changes
+  themeUpdated: EventEmitter<Theme> = new EventEmitter();
```

Comment:
I think ThemeService should take ownership of (un)subscribing events.

EventEmitter should be an implementation detail (that we might change it in the future). So in ThemeService, you would expose (un)subscribe() methods to clients and use an instance of EventEmitter to do the actual work (subscribing).

Reply:
nit: "themeEventHandler" may be a better name than "themeUpdated".

Reply:
https://angular.io/guide/component-interaction#parent-and-children-communicate-using-a-service

According to this, it is much better to use `rxjs/Subject` rather than `EventEmitter`. I think this solves the purpose of abstracting the `emit` method to a function and provide only the observable stream object as public.

Reply:
I've replaced EventEmitter with the Subject like Angular. Now components are not allowed to emit anything, they can only subscribe to the observable exposed by the service.